zstackctl/zstackctl/ctl.py

```
@@ -177,7 +177,31 @@
+#      leave my.cnf pointing at a missing file -> mariadbd aborts on next start.
+if grep -qi 'Alibaba Cloud Linux' /etc/os-release 2>/dev/null; then
+    grep 'init-file=.*init_fk.sql' $mysql_conf >/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        echo "foreign_key_checks=0 via init-file (Alinux4 workaround)"
+        init_fk_dir=/var/lib/zstack
+        init_fk_file=$init_fk_dir/init_fk.sql
+        mkdir -p $init_fk_dir
+        echo "SET GLOBAL foreign_key_checks=0;" > $init_fk_file
+        chown mysql:mysql $init_fk_file
+        chmod 644 $init_fk_file
+        sed -i "/\[mysqld\]/a init-file=$init_fk_file" $mysql_conf
```

Ruff (0.15.12)

[warning] Invalid escape sequence: `\[`

Add backslash to escape sequence

(W605)